projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/authorization/KerberosAuthenticationIT.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.vmware.taurus.authorization;
+
+import com.kerb4j.client.SpnegoClient;
+import com.kerb4j.client.SpnegoContext;
+import com.vmware.taurus.ControlplaneApplication;
+import com.vmware.taurus.service.JobsRepository;
+import com.vmware.taurus.service.model.DataJob;
+import com.vmware.taurus.service.model.JobConfig;
+import org.apache.kerby.kerberos.kerb.KrbException;
+import org.apache.kerby.kerberos.kerb.server.SimpleKdcServer;
+import org.apache.kerby.util.NetworkUtil;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.io.File;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * This test is in the unit tests directory, because
```

Comment:
I would say the test is in the unit test because 
A) it's fast (it takes a few seconds)
B) It mocks and manages all its dependencies and does not create side effects . 
C) It does not work well with MiniKDC 

The Mini KDC related issues - I would add as a TODO in teh MniKDC class in the integration test suite

But even if we replace MiniKDC with Kerby it's fine to leave the test here. for reasons A and B.